### ISSUE TITLE
autojack.json not present bug

### ISSUE THREAD

**Issue by Elkenfugel** [ovenwerks/studio-controls#46]
1. Ran studio-controls and  dropped a .json error

```
bash:~$ studio-controls

Config file: /home/pc/.config/autojack/autojack.json
Traceback (most recent call last):
  File "/usr/bin/studio-controls", line 1420, in <module>
    us = StudioControls()
  File "/usr/bin/studio-controls", line 437, in __init__
    with open(c_file) as f:
FileNotFoundError: [Errno 2] No such file or directory: '/home/pc/.config/autojack/autojack.json'
pc@pc:~$ autojack
starting up
sending quit to any old autojack
Lock file created
logging started
Traceback (most recent call last):
  File "/usr/bin/autojack", line 1412, in <module>
    main()
  File "/usr/bin/autojack", line 1357, in main
    phones_check()
  File "/usr/bin/autojack", line 1201, in phones_check
    import_config("phones_check")
  File "/usr/bin/autojack", line 246, in import_config
    with open(c_file) as f:
FileNotFoundError: [Errno 2] No such file or directory: '/home/pc/.config/autojack/autojack.json'
```

2. did convert-studio-controls it failed
3. installed ubuntu_studio version via APT
4. ran convert-studio-controls
5. ensured the autojackrc file was present
6. _went to line 155 and added **version = 2.1_**
7. got a json file (attached at end)
8. STARTED STUDIO CONTROLS FROM THE COMMAND LINE !!! (i am excited because its taken me a couple of hours....like 4 hours hahaha)
9. got firefox audio running through jack to pulse to usb blue yeti. 

```
PC@pc:~/.config/autojack$ convert-studio-controls
read old file
make new data base
Traceback (most recent call last):
  File "/usr/bin/convert-studio-controls", line 380, in <module>
    main(sys.argv[1:])
  File "/usr/bin/convert-studio-controls", line 376, in main
    make_db()
  File "/usr/bin/convert-studio-controls", line 157, in make_db
    our_db = {'version': version}
NameError: name 'version' is not defined
elken@curnelius:~/.config/autojack$ convert-studio-controls
  File "/usr/bin/convert-studio-controls", line 155
    version = 2.1.0
                 ^
SyntaxError: invalid syntax


```
```

{
    "version": 2.1,
    "log-level": 15,
    "jack": {
        "on": false,
        "driver": "alsa",
        "chan-in": 0,
        "chan-out": 0,
        "rate": 48000,
        "frame": 1024,
        "period": 2,
        "connect-mode": "n",
        "dev": "PCH,0,0",
        "usbdev": ""
    },
    "extra": {
        "a2j": true,
        "usbauto": true,
        "usb-single": false,
        "monitor": "system:playback_1",
        "phone-action": "switch",
        "phone-device": "PCH,0,0"
    },
    "pulse": {
        "inputs": {
            "pulse_in": {
                "connection": "system:capture_1",
                "count": 2
            }
        },
        "outputs": {
            "pulse_out": {
                "connection": "monitor",
                "count": 2
            }
        }
    },
    "devices": {
        "PCH": {
            "number": -1,
            "usb": false,
            "internal": true,
            "hdmi": false,
            "rates": [
                "32000",
                "44100",
                "48000",
                "88200",
                "96000",
                "192000"
            ],
            "sub": {
                "0": {
                    "playback": true,
                    "capture": true,
                    "play-chan": 0,
                    "cap-chan": 0,
                    "rate": 48000,
                    "frame": 1024,
                    "nperiods": 2,
                    "hide": false
                }
            }
        }
    }
}

```

**Comment by ovenwerks** [comment]
{
    "version": 2.1,
Hmm it appears this is a hand edited file. There is no such version as 2.1 and the version is always a string. This would be imported as a number which when compared to the string version would throw the error you have shown. Ah, you: "went to line 155 and added **version = 2.1**" That should be version = "2.1.0"


I think this extends from the other bugs you have given (not seeing local and perhaps not installing correctly). I am working on those now.

**Comment by Elkenfugel** [comment]
yes i had to define varible version as the autojackrc config file some how was completey gone.

the package version (i forget which version) of studio-controls broke and would not run or start jack.

im not sure how i broke it but it stopped working completey. hence the need for me to remove and install the package version alongside the current release (2.1.0), run convert json script then remove the ubuntu package version, do the other sutudio-controls/line 290 fix for use/share and move everything to /usr/bin (not /usr/local/bin) then i got a working build

so the whole "make a default config on first run" block of code did not work for me at all... i had to manually do it.

yes i manually had to define that varible then it was able to make the .json



Valentino Ferriera
________________________________

**Comment by ovenwerks** [comment]
This should be fixed in 2.1.1
